lib/rules/no-this-before-super.js

```diff
@@ -197,11 +197,47 @@ module.exports = {
                     return;
                 }
 
-                codePath.traverseSegments((segment, controller) => {
+                /**
+                 * A collection of nodes to avoid duplicate reports.
+                 * @type {Set<ASTNode>}
+                 */
+                const reported = new Set();
+
+                /**
+                 * A stack of segments to visit.
+                 * @type {CodePathSegment[]}
+                 */
+                const segmentStack = [codePath.initialSegment];
+
+                /**
+                 * A collection of segments to avoid duplicate visits.
+                 * @type {Set<CodePathSegment>}
+                 */
+                const visited = new Set();
+
+                /*
+                 * Traverse the segments in order starting from the first segment
+                 * and check for `this` used before `super()`.
+                 */
+                while (segmentStack.length > 0) {
```

Comment:
Should we maybe try to fix `CodePath#traverseSegments()` instead of reimplementing it here?

Reply:
I changed this PR to fix traverseSegments 👍